UnityProject/Assets/Scripts/Objects/Construction/GrilleObject.cs

```diff
@@ -0,0 +1,113 @@
+		return true;
+	}
+
+	public void ServerPerformInteraction(HandApply interaction)
+	{
+		if (interaction.TargetObject != gameObject) return;
+
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Screwdriver))
+		{
+			if (objectBehaviour.IsPushable)
+			{
+				//secure it if there's floor
+				if (MatrixManager.IsSpaceAt(registerObject.WorldPositionServer, true))
+				{
+					Chat.AddExamineMsg(interaction.Performer, "A floor must be present to secure the grille!");
+					return;
+				}
+
+				if (!ServerValidations.IsAnchorBlocked(interaction))
+				{
+					ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+						"You start securing the grille...",
+						$"{interaction.Performer.ExpensiveName()} starts securing the grille...",
+						"You secure the grille.",
+						$"{interaction.Performer.ExpensiveName()} secures the grille.",
+						() => ScrewInPlace(interaction));
+					SoundManager.PlayNetworkedAtPos("screwdriver2", gameObject.TileWorldPosition().To3Int(), Random.Range(0.9f, 1.1f));
+				}
+			}
+			else
+			{
+				//unsecure it
+				ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+					"You start unsecuring the grille...",
+					$"{interaction.Performer.ExpensiveName()} starts unsecuring the grille...",
+					"You unsecure the grille.",
+					$"{interaction.Performer.ExpensiveName()} unsecures the grille.",
+					() => objectBehaviour.ServerSetAnchored(false, interaction.Performer));
+				SoundManager.PlayNetworkedAtPos("screwdriver2", gameObject.TileWorldPosition().To3Int(), Random.Range(0.9f, 1.1f));
+			}
+
+		}
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wirecutter))
+		{
+
+				ToolUtils.ServerUseToolWithActionMessages(interaction, 1f,
+					"You start to disassemble the grille...",
+					$"{interaction.Performer.ExpensiveName()} starts to disassemble the grille...",
+					"You disassemble the grille.",
+					$"{interaction.Performer.ExpensiveName()} disassembles the grille.",
+					() => Disassemble(interaction));
+			SoundManager.PlayNetworkedAtPos("wirecutter", gameObject.TileWorldPosition().To3Int(), Random.Range(0.9f, 1.1f));
+		}
+	}
+
+	[Server]
+	private void Disassemble(HandApply interaction)
+	{
+		Spawn.ServerPrefab(CommonPrefabs.Instance.MetalRod, registerObject.WorldPositionServer, count: 2);
+		GetComponent<CustomNetTransform>().DisappearFromWorldServer();
+	}
+
+	[Server]
+	private void ScrewInPlace(HandApply interaction)
+	{
+		var interactableTiles = InteractableTiles.GetAt(interaction.TargetObject.TileWorldPosition(), true);
+		Vector3Int cellPos = interactableTiles.WorldToCell(interaction.TargetObject.TileWorldPosition());
+		interactableTiles.TileChangeManager.UpdateTile(cellPos, layerTile);
+		interactableTiles.TileChangeManager.SubsystemManager.UpdateAt(cellPos);
+		GetComponent<CustomNetTransform>().DisappearFromWorldServer();
```

Comment:
Also this is the same code as in window. Is it better if I make it as in Girder?

Reply:
no, you don't need `SubsystemManager.UpdateAt` since it's not airtight anyway, no need to update atmos.
you'll still need to despawn this object though, check previous comment